cmd/lint.go

```diff
@@ -396,8 +398,8 @@ func getWriterForOutputFile(filename string) (io.Writer, error) {
 }
 
 func formatError(format string, err error) error {
-	// currently, JSON and SARIF will get the same generic JSON error format
-	if format == formatJSON || format == formatSarif {
+	// currently, JSON, SARIF, and JUnit will get the same generic JSON error format
+	if format == formatJSON || format == formatSarif || format == formatJunit {
```

Comment:
I would probably have expected the errors to be reported as "JUnit errors"... but I guess we'd need to update the reporters to allow for that. We'll deal with that later.

Reply:
OK thanks for pinging me about this - I wasn't really sure why this code exists. In my understanding, this function is only called if the lint command itself failed. As a user I would not expect to get any kind of formatted report for that but rather a harsh error message telling me to fix my setup before regal can work properly.

That said, I could add proper JUnit XML output here as well. Besides `Failure` it does contain an `Error` field that could be used here..

Reply:
Yeah, but the lint command can fail for reasons like parser errors too. Ideally, other tools like `opa test` would have failed the build before linting in that case, but still. In most of the contexts where e.g. SARIF is used, like in CI/CD pipelines, users would likely want to find the report in the same place regardless of what made it fail, but yeah, there's a reason we've not exactly made this our highest priority too.

Reply:
OK OK sounds good - will add JUnit output here as well!

Reply:
If non-trivial to pull off, it's alright to defer this to later. We should look into better error reporting altogether sometime soon anyway.

Reply:
I think the latest version is a reasonable start - it basically just creates a single testsuite with a single testcase that contains the given error string.